src/lib/spinel/spinel.h

```diff
@@ -4836,6 +4836,38 @@ enum
      */
     SPINEL_PROP_DNSSD_HOST = SPINEL_PROP_DNSSD__BEGIN + 3,
 
+    /// DNS-SD Service
+    /**
+     * Format `UUUSA(U)SdLD`: Inserted/Removed
+     *
+     * `U`    : The host name (does not include domain name).
+     * `U`    : The service instance name label (not the full name).
+     * `U`    : The service type (e.g., "_mt._udp", does not include domain name).
+     * `S`    : The count of sub-type labels.
+     * `A(U)` : Array of sub-type labels (can be NULL if no label).
+     * `S`    : The service port number.
+     * `d`    : Encoded TXT data bytes.
+     * `L`    : The Dnssd Request ID.
+     * `D`    : The context of the request. (A pointer to the callback for the request)
+     *
+     * NCP uses this property to register/unregister a DNS-SD service.
+     */
+    SPINEL_PROP_DNSSD_SERVICE = SPINEL_PROP_DNSSD__BEGIN + 4,
+
+    /// DNS-SD Key Record
+    /**
+     * Format `UUdLD`: Inserted/Removed
+     *
+     * `U` : A host or a service instance name (does not include domain name).
+     * `U` : The service type if key is for a service (does not include domain name).
```

Comment:
We should be careful here, in the `otPlatDnssdKey` struct, we use the fact that `mServiceType` is `NULL` to distinguish name format (service instance vs host name). We cannot do the same here. 

Some options:
- Can use `""` string 
- Can wrap the whole `U` in a struct like `S(U)`. When parsing, we can tell if `S` is empty, it indicates null 

I like the second way better (as it mirrors notion of `NULL` better and I think may be easier to parse/decode and encode (spinel prespective).

Reply:
I also prefer using struct.